C#/C#.sublime-syntax

```diff
@@ -1,276 +1,1593 @@
-        - meta_scope: string.quoted.single.source.cs
-        - match: "'"
-          scope: punctuation.definition.string.end.source.cs
+        - match: ':'
+          scope: keyword.operator.ternary.cs
           pop: true
-        - match: \\.
-          scope: constant.character.escape.source.cs
-  keywords:
-    - match: \b(if|else|while|for|foreach|in|do|return|continue|break|switch|case|default|goto|throw|try|catch|finally|lock|yield)\b
-      scope: keyword.control.source.cs
-    - match: \b(from|where|select|group|into|orderby|join|let|on|equals|by|ascending|descending)\b
-      scope: keyword.linq.source.cs
-    - match: \b(new|is|as|using|checked|unchecked|typeof|sizeof|override|readonly|stackalloc)\b
-      scope: keyword.operator.source.cs
-    - match: \b(var|event|delegate|add|remove|set|get|value)\b
-      scope: keyword.other.source.cs
-  method:
-    - include: attributes
-    - match: '(?=\bnew\s+)(?=[\w<].*\s+)(?=[^=]+\()'
+        - include: line_of_code_in
+
+    - match: '(\()\s*({{base_type}}|{{name}})\s*((?:\[,*\]|\*)*)\s*(\))'
+      captures:
+        1: punctuation.definition.arguments.casting.begin.cs
+        2: support.type.cs
+        3: variable.other.type.cs
+        4: support.type.cs
+        5: punctuation.definition.arguments.casting.end.cs
+
+    - match: \(
+      scope: punctuation.definition.expression.begin.cs
       push:
-        - meta_scope: meta.new-object.source.cs
-        - match: "(?={|;)"
+        - match: \)
+          scope: punctuation.definition.expression.end.cs
+          pop: true
+        - match: ','
+          scope: punctuation.separator.expression.cs
+        - include: line_of_code_in
+    - match: \{
+      scope: punctuation.section.block.begin.cs
+      set:
+        - match: \}
+          scope: punctuation.section.block.end.cs
           pop: true
-        - include: code
-    - match: '(?!new)(?=[\w<].*\s+)(?=[^=]+\()'
+        - include: code_block_in
+    - match: ';'
+      scope: punctuation.separator.statement.cs
+      pop: true
+    - match: (?=\}|\)|>|\]|,)
+      pop: true
+
+  arguments:
+    - match: (out|ref)\s
+      scope: storage.modifier.argument.cs
+    - match: '{{name}}\s*(:)'
+      captures:
+        1: variable.parameter.cs
+        2: punctuation.separator.argument.value.cs
+      push: line_of_code_in
+    - match: ','
+      scope: punctuation.separator.argument.cs
+    - match: \)
+      scope: punctuation.definition.arguments.end.cs
+      pop: true
+    - match: '(?=\S)'
+      push: line_of_code_in
+
+  accessor_arguments:
+    - match: '{{name}}\s*(:)'
+      captures:
+        1: variable.other.parameter.cs
+        2: punctuation.separator.key-value.parameter.cs
+    - match: ','
+      scope: punctuation.separator.accessor.cs
+    - match: \]
+      scope: punctuation.definition.accessor.end.cs
+      pop: true
+    - include: line_of_code_in
+
+  type:
+    - include: type_common
+    - match: '{{name}}'
+      scope: variable.other.type
```

Comment:
this is missing the `.cs` suffix. There are a few other instances of `scope: variable.other.type` where this applies too.